x/emissions/keeper/inference_synthesis/network_inferences.go

```diff
@@ -1,200 +1,242 @@
-		if err != nil || len(inferences.Inferences) == 0 {
-			return nil, nil, nil, nil, inferenceBlockHeight, lossBlockHeight, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "no inferences found for topic %v at block %v", topicId, *inferencesNonce)
+		inferences, err := k.GetInferencesAtBlock(ctx, topicId, *inferencesNonce)
+		if err != nil {
+			return nil, 0, err
+		} else if len(inferences.Inferences) == 0 {
+			return nil, 0, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "no inferences found for topic %v at block %v", topicId, *inferencesNonce)
 		}
+		return inferences, *inferencesNonce, nil
+	}
+}
+
+func calcNetworkInferencesMultipleByMedian(
+	ctx sdk.Context,
+	topicId TopicId,
+	inferences *emissions.Inferences,
+	inferenceBlockHeight BlockHeight,
+) (*GetNetworkInferencesResult, error) {
+	inferenceValues := fn.Map(inferences.Inferences, func(inf *emissions.Inference) alloraMath.Dec { return inf.Value })
+
+	medianValue, err := alloraMath.Median(inferenceValues)
+	if err != nil {
+		return nil, errorsmod.Wrap(err, "while calculating median")
 	}
 
-	// ATTN: PROTO-2464
-	networkInferences = &emissions.ValueBundle{
+	networkInferences := &emissions.ValueBundle{
 		TopicId: topicId,
 		ReputerRequestNonce: &emissions.ReputerRequestNonce{
 			ReputerNonce: &emissions.Nonce{BlockHeight: ctx.BlockHeight()},
 		},
-		Reputer:                       "allo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqas6usy",
-		ExtraData:                     nil,
-		CombinedValue:                 alloraMath.ZeroDec(),
-		InfererValues:                 make([]*emissions.WorkerAttributedValue, 0),
-		ForecasterValues:              make([]*emissions.WorkerAttributedValue, 0),
+		Reputer:       "allo1qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqas6usy",
+		CombinedValue: medianValue,
+		InfererValues: fn.Map(inferences.Inferences, func(inf *emissions.Inference) *emissions.WorkerAttributedValue {
+			return &emissions.WorkerAttributedValue{Worker: inf.Inferer, Value: inf.Value}
+		}),
+		ForecasterValues:              make([]*emissions.WorkerAttributedValue, 0), // TODO(spook): can all of these be nil?
 		NaiveValue:                    alloraMath.ZeroDec(),
-		OneOutInfererValues:           make([]*emissions.WithheldWorkerAttributedValue, 0),
-		OneOutForecasterValues:        make([]*emissions.WithheldWorkerAttributedValue, 0),
-		OneInForecasterValues:         make([]*emissions.WorkerAttributedValue, 0),
-		OneOutInfererForecasterValues: make([]*emissions.OneOutInfererForecasterValues, 0),
+		OneOutInfererValues:           make([]*emissions.WithheldWorkerAttributedValue, 0), // TODO(spook): can all of these be nil?
+		OneOutForecasterValues:        make([]*emissions.WithheldWorkerAttributedValue, 0), // TODO(spook): can all of these be nil?
+		OneInForecasterValues:         make([]*emissions.WorkerAttributedValue, 0),         // TODO(spook): can all of these be nil?
+		OneOutInfererForecasterValues: make([]*emissions.OneOutInfererForecasterValues, 0), // TODO(spook): can all of these be nil?
```

Comment:
only thing that occurred to me is that we might want to prevent the json representation from being null (or entirely missing) -- we can have gogoprotobuf remove the `,omitempty` struct tag if so

Reply:
removed